Il2CppInterop.Runtime/Injection/ClassInjector.cs

```diff
@@ -198,8 +205,8 @@ public static void RegisterTypeInIl2Cpp(Type type, RegisterTypeOptions options)
         // Initialize the vtable of all base types (Class::Init is recursive internally)
         InjectorHelpers.ClassInit(baseClassPointer.ClassPointer);
 
-        if (baseClassPointer.ValueType || baseClassPointer.EnumType)
-            throw new ArgumentException($"Base class {baseType} is value type and can't be inherited from");
```

Comment:
Does `baseClassPointer.ValueType` return `true` for `System.ValueType`? If not, this shouldn't be changed.

Reply:
I'm pretty sure it does return `true`. This check being removed is exactly why structs can be injected. This is big part of this PR.

Specifically structs are needed because Unity's ECS (Entity Component System) heavily relies on structs to hold game state. By injecting custom ones we can add our own.